includes/admin/reporting/tables/llms.table.students.php

```diff
@@ -186,7 +188,12 @@ public function get_data( $key, $student ) {
 					$value = $student->get( 'last_login' );
 				}
 
-				$value = $value ? date_i18n( get_option( 'date_format' ), is_numeric( $value ) ? $value : strtotime( $value ) ) : '&ndash;';
+				if ( $value ) {
+					$timestamp = is_numeric( $value ) ? $value : strtotime( $value );
+					$value     = wp_date( get_option( 'date_format' ), $timestamp );
```

Comment:
@pondermatic 
I'm not convinced about this.

There is a problem, but I'm not convinced it's the one you're describing and solving here.

The thing is that that column basically shows this value:
https://github.com/gocodebox/lifterlms/blob/5.2.0/includes/abstracts/llms.abstract.database.store.php#L115
which is basically `current_time()` in `'mysql'` format  plus the wp timezone offset:
https://github.com/gocodebox/lifterlms/blob/dae9300e19b5cb2cd48458e88cd8348ca1a2274b/includes/llms.functions.core.php#L94
So here we're replacing [date_i18n()](https://developer.wordpress.org/reference/functions/date_i18n/) wit [wp_date()](https://developer.wordpress.org/reference/functions/wp_date/).
The first expects a timestamp with offset (timezone), while the second expects a timestamp without timezone.
The thing is that our saved value is a timestamp with offset, but the offset of course is calculated with the wp timezone at the moment the entry is created.
So using `wp_date` here is not an option, I think...

I have to think a little more about this but I'm not sure there's a solution, if not converting (with a db update) all those saved dates to UTC (with a guess) and then go with `wp_date`

Reply:
You are correct @eri-trabiccolo, [`LLMS_Event` dates](https://github.com/gocodebox/lifterlms/blob/5.2.0/includes/abstracts/llms.abstract.database.store.php#L115), stored in the `wp_lifterlms_events` table, are relative to the local time zone. The [`last_login`](https://github.com/gocodebox/lifterlms/blob/5.2.0/includes/functions/llms.functions.person.php#L487) time, stored in `wp_usermeta` with the `llms_last_login` key, is relative to the UTC time zone.

I'll change `LLMS_Table_Students::get_data()` to handle both values and also add a test using an event.

Reply:
@pondermatic 
but actually the `last_login` time is in local time too, no?
https://github.com/gocodebox/lifterlms/blob/5.2.0/includes/abstracts/llms.abstract.database.store.php#L115
https://github.com/gocodebox/lifterlms/blob/5.2.0/includes/functions/llms.functions.person.php#L487
both are using `llms_current_time('mysql');`